Lib/defcon/objects/glyph.py

```diff
@@ -838,13 +838,12 @@ def insertAnchor(self, index, anchor):
 
         This will post a *Glyph.Changed* notification.
         """
-        try:
-            assert anchor.glyph is None
-        except AttributeError:
-            pass
+        for anchor_ in self._anchors:
+            assert anchor is not anchor_
```

Comment:
why is it so important to prevent the same anchor instance being inserted more than once?

Reply:
and anyway, doesn't the identifier check below accomplish the same thing (prevent anchors with same identifier)?

Reply:
> why is it so important to prevent the same anchor instance being inserted more than once?

It will cause problems with notifications, see: https://github.com/typesupply/defcon/pull/85#issuecomment-239583706 (it would have duplicate key in the notifications dict when subscribing)

So here as I'm allowing anchors with glyph=self I'm also adding this check (such case would be blocked previously).

> and anyway, doesn't the identifier check below accomplish the same thing (prevent anchors with same identifier)?

Identifiers are optional, they are not generated by default.

Reply:
thanks for the clarification